File: Matlab/+ACME/Color/position2color.m
function [C] = position2color(P,Min,Max)
% POSITION2COLOR  Converts a 3-dimensional coordinate into a color.
%   C = POSITION2COLOR(P) returns a nx3 vector, with C(i,:) representing
%   the color P(i,:).
%   C = POSITION2COLOR(P,Min) returns a nx3 vector, where the color black
%   represents the coordinate Min
%   C = POSITION2COLOR(P,Min,Max) returns a nx3 vector, where the color
%   black represents the coordinate Min, while white represents Max
if( nargin < 3 )
    Max = max(P);
end
if( nargin < 2 )
    Min = min(P);
end
C = normalize(clamp3(P,Min,Max),Min,Max);
end

File: encoding/position2color.py
import torch
from ..math.normalize import *


def position2color(P, min=None, max=None):
    """
    Converts the input points set into colors

    Parameters
    ----------
    P : Tensor
        the input points set tensor
    min : float (optional)
        the minimum value for the points set. If None it will be automatically computed (default is None)
    max : float (optional)
        the maximum value for the points set. If None it will be automatically computed (default is None)

    Returns
    -------
    Tensor
        the color tensor
    """

    if min is None:
        min = torch.min(P)[0].item()
    if max is None:
        max = torch.max(P)[0].item()
    return normalize(P, min=min, max=max)
